Source/Scene/Primitive.js

```diff
@@ -947,6 +991,36 @@ define([
         return [attributeDecl, globalDecl, modifiedVS, compressedMain].join('\n');
     }
 
+    function depthClampVS(vertexShaderSource) {
+        var modifiedVS = ShaderSource.replaceMain(vertexShaderSource, 'czm_non_depth_clamp_main');
+        modifiedVS += 'varying float v_WindowZ;\n' +
+                      'void main() {\n' +
+                      '    czm_non_depth_clamp_main();\n' +
+                      '    vec4 position = gl_Position;\n' +
+                      '    v_WindowZ = (0.5 * (position.z / position.w) + 0.5) * position.w;\n' +
+                      '    position.z = min(position.z, position.w);\n' +
+                      '    gl_Position = position;' +
+                      '}\n';
+        return modifiedVS;
+    }
+
+    function depthClampFS(context, fragmentShaderSource) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!context.fragmentDepth) {
+            throw new DeveloperError('The depth fail appearance requires the EXT_frag_depth extension.');
+        }
+        //>>includeEnd('debug');
+
+        var modifiedFS = ShaderSource.replaceMain(fragmentShaderSource, 'czm_non_depth_clamp_main');
+        modifiedFS += 'varying float v_WindowZ;\n' +
+              'void main() {\n' +
+              '    czm_non_depth_clamp_main();\n' +
+              '    gl_FragDepthEXT = min(v_WindowZ * gl_FragCoord.w, 1.0);\n' +
```

Comment:
Could you get a screenshot of that happening?

It may be worth supporting a fallback shader when the extension isn't supported, since it still mostly works without.